src/sfctl/tests/help_text_test.py

```diff
@@ -0,0 +1,364 @@
+
+            return (subgroups_index, commands_index)
+
+        elif (section == 'Commands') or (section == 'Subgroups'):
+            # Make sure that if the line starts with the command/group in
+            # the expected tuple, that a description follows it.
+            # The line will either start with the name provided in the expected tuple,
+            # or it will be a continuation line.
+            first_word_of_line = line.split()[0].rstrip(':')
+
+            if section == 'Commands':
+
+                # If we've reached the end of the commands tuple, then skip
+                if len(commands) == commands_index:
+                    return (subgroups_index, commands_index)
+
+                self.assertGreater(len(commands), commands_index,
+                                   msg='None or missing expected commands provided in test for ' + command_input) #pylint: disable=line-too-long
+                if first_word_of_line == commands[commands_index]:
+                    # make sure there is descriptive text in this line
+                    first_index = line.find(': ')
+                    self.assertGreater(
+                        len(line),
+                        len(first_word_of_line),
+                        msg='Validating help text failed in "Commands" on line: ' + line)
+
+                    commands_index += 1
+
+            elif section == 'Subgroups':
+
+                # If we've reached the end of the commands tuple, then skip
+                if len(subgroups) == subgroups_index:
+                    return (subgroups_index, commands_index)
+
+                self.assertGreater(len(subgroups), subgroups_index,
+                                   msg='None or missing expected subgroups provided in test for ' + command_input) #pylint: disable=line-too-long
+                if first_word_of_line == subgroups[subgroups_index]:
+                    # make sure there is descriptive text in this line
+                    first_index = line.find(': ')
+                    self.assertGreater(
+                        len(line),
+                        len(first_word_of_line),
+                        msg='Validating help text failed in "Subgroups" on line: ' + line)
+
+                    subgroups_index += 1
+
+            return (subgroups_index, commands_index)
+
+        else:
+            self.fail('Section name {0} is not supported'.format(section))
+
+    @classmethod
+    def _validate_output_read_section_name(cls, line):
+        """ Read a given line and validate it for correctness based on the given section.
+        Parameter line (string) should be unprocessed. For example, the line should not be
+        stripped of starting or trailing white spaces.
+
+        Returns the section name if the given line designates the beginning of a new section.
+        Returns None if the line does not. """
+
+        if (line.strip() and (line[0] != ' ')):
```

Comment:
I think this will fail if the line starts with any whitespace that is not a space. Try to simplify this logic. I think you want `lstrip` and `rstrip` here.